clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```
@@ -90,14 +93,13 @@ public void run() {
             while (running) {
                 try {
                     runOnce();
-                } catch (final WakeupException e) {
-                    log.debug("WakeupException caught, consumer network thread won't be interrupted");
+                } catch (final Throwable e) {
+                    log.error("Unexpected error caught in consumer network thread", e);
                     // swallow the wakeup exception to prevent killing the thread.
```

This comment is now outdated.

---

Thanks, a question here: Should we actually swallow the exception or propagate the exception back to the application thread (wrap it with KafkaException via BackgroundEventHandler)

---

Good question. I'm not sure what will end up in the catch block here. The code around processing events seems to be wrapped by another catch-all block, so we'd expect only problems around sending requests to end up here? If you are not sure, open a ticket for investigation.